scala/scalafmt/BUILD

```diff
@@ -0,0 +1,37 @@
+load("//scala:scala.bzl", "scala_binary")
+
+filegroup(
+    name = "runner",
+    srcs = ["private/format.template.sh"],
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "testrunner",
+    srcs = ["private/format-test.template.sh"],
+    visibility = ["//visibility:public"],
+)
+
+scala_binary(
+    name = "scalafmt",
+    srcs = glob(["scalafmt/ScalafmtRunner.scala"]),
+    main_class = "io.bazel.rules_scala.scalafmt.ScalafmtRunner",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//external:io_bazel_rules_scala/scalafmt/argparse4j",
+        "//external:io_bazel_rules_scala/scalafmt/geirsson_metaconfig_core",
+        "//external:io_bazel_rules_scala/scalafmt/scalafmt_parsers",
+        "//src/java/io/bazel/rulesscala/worker",
+        "@scalafmt//:org_scalameta_scalafmt_core_2_11",
+    ],
+)
+
+load(
+    "//scala/scalafmt:phase_scalafmt_ext.bzl",
+    "scalafmt_singleton",
+)
+
+scalafmt_singleton(
+    name = "phase_scalafmt",
```

Comment:
what does this name mean?

Reply:
It's a phase to format Scala source code by [scalafmt](https://scalameta.org/scalafmt/). I believe `fmt` is shortened from `format.

Reply:
I'm sorry I wasn't clear. I meant I don't understand the scalafmt_singleton pattern and also what does then name mean (in the bazel concept, not literaly)

Reply:
To be honest, I don't have answer. It comes from the convention of `higherkindness/rules_scala`. I personally interpret it as a dummy rule just to pass in `_ScalaRulePhase` provider.

We don't have to follow any convention. I lean toward setting up principle for future reference. Unless @andyscott can provide more insight here.